src/python/pants/engine/internals/selectors.py

```diff
@@ -699,14 +698,9 @@ def __init__(self, *args: Any) -> None:
         self.params = tuple(args)
 
 
-_RAISE_KEYBOARD_INTERRUPT = os.environ.get("_RAISE_KEYBOARD_INTERRUPT_FFI", None)
-
-
 def native_engine_generator_send(
     func, arg
 ) -> PyGeneratorResponseGet | PyGeneratorResponseGetMulti | PyGeneratorResponseBreak:
-    if _RAISE_KEYBOARD_INTERRUPT:
-        raise KeyboardInterrupt("ctrl-c interrupted execution during FFI (for testing purposes).")
```

Comment:
This synthetically raised `KeyboardInterrupt` will not actually trigger the rust-side signal handler, and so was testing an effectively impossible case.